src/main.cpp

```diff
@@ -59,7 +61,41 @@ void sokoban_initialize() {
             "######",
         },
     };
-    soko = {levels};
+    soko = {levels};*/
+    //std::vector <std::vector<std::string>> levels;
+    std::filesystem::path getLevelsDir = std::filesystem::directory_entry("dist/levels");
+    std::vector<std::filesystem::path> getMapPath; // save path into vector element
+
+    for (const auto& entry : std::filesystem::directory_iterator(getLevelsDir)) {
+        getMapPath.push_back(entry.path());
+    }
+
+    // removes every other instance of a file found in the directory, namely the .data file since they are every other in the directory, to leave only .txt files in the vector
+    for (unsigned int i = 0; i < getMapPath.size(); i++) {
+        getMapPath.erase(getMapPath.begin() + i);
+    }
+
+    // opening map files stored in vector directory
+    for (unsigned int i = 0; i < getMapPath.size(); i++) {
+        std::ifstream openMapFile(getMapPath[i].c_str());
+        std::string store_line;
+        std::vector<std::string> storeMap;
+
+        if (!openMapFile) {
+            std::cerr << "Cannot open the File : " << getMapPath[i] << std::endl;
+        }
+
+        for (unsigned int i = 0; i < 1; i++) {
+            while (std::getline(openMapFile, store_line)) {
+                if (store_line.find("#") == false || store_line.find(" ") == false) {
```

Comment:
We should be able to say something like "if this line is entirely comprised of Sokoban characters, parse it, otherwise ignore it". Regex might be good for this. We don't have to try too hard to make it robust, so this is OK for now though.

I'd use `!store_line.find("#")` rather than `== false`.